### PR TITLE
Fix/app fetcher php compat comparison

### LINKED ISSUE #4383: Mail app not available for installation
### Expected behavior
The mail app must show up in available apps from the Nextcloud web interface

### Actual behavior
The app is not liste

### Mail app

NA

### Server configuration

**Operating system**: Debian 10

**Web server:** Apache

**Database:** MariaDB

**PHP version:** 7.3.19-1

**Nextcloud Version:** 20.0.5

#### Client configuration
**Browser:** Firefox 84.0.2 (64-bit)

**Operating system:** Ubuntu 20.10

Nextcloud Report attached
[2021-01-22.txt](https://github.com/nextcloud/mail/files/5855266/2021-01-22.txt)

### PR DESCRIPTION
**Pull request by ChristophWurst**
Reverts bedd9acf7874b836bef224989c2ce7f5e3b0a2fa.

This is how https://github.com/nextcloud/server/pull/24416 was meant to work.

Fixes https://github.com/nextcloud/mail/issues/4383

Note to testers: there is some caching that might make it impossible to see the fix immediately. Apply the changes as patch from https://patch-diff.githubusercontent.com/raw/nextcloud/server/pull/25335.patch, give the cache an hour or so to expire and try again

### CODE REVIEW

**Review by kesselb — APPROVED** [review]
LGTM

File: tests/lib/App/CompareVersionTest.php [hunk 1]
```diff
@@ -49,11 +51,17 @@ public function comparisonData() {
 			['13.0.0', '13', '>=', true],
 			['13.0.1', '13', '>=', true],
 			['13.0.1', '13', '<=', true],
+			['13.0.1.9', '13', '<=', true],
+			['13.0.1-beta.1', '13', '<=', true],
+			['7.4.14', '7.4', '<=', true],
+			['7.4.14-ubuntu', '7.4', '<=', true],
+			['7.4.14-ubuntu', '7.4.15', '<=', true],
```
Comment (nickvergessen) [inline tests/lib/App/CompareVersionTest.php:58]: Can we also have a false test with Ubuntu

**Review by nickvergessen — APPROVED** [review]

**Comment by szaimen** [comment]
Will this get backported to NC20?

File: tests/lib/App/CompareVersionTest.php (hunk 1 above)
Comment (ChristophWurst) [reply tests/lib/App/CompareVersionTest.php:58]: https://github.com/nextcloud/server/pull/25348

**Comment by ManafxMT** [comment]
Hello Christoph
I have the same problem - Email disappered since my update to 20.0.6.

How to implement the patch please?
Do you have a link where I can read about the procedure?

I never had an issue like this before.

**Comment by ChristophWurst** [comment]
Let me show you a magic trick. The PR URL is `https://github.com/nextcloud/server/pull/25335`. Now append a `.patch` and you will be redirected to https://patch-diff.githubusercontent.com/raw/nextcloud/server/pull/25335.patch. Download that patch.

Then you can use your `patch` command to apply the changes: https://www.cyberciti.biz/faq/appy-patch-file-using-patch-command/.

**Comment by vladlucky** [comment]
Having the same issue as described here by others - unable to locate/enable the TOTP app in Nextcloud. 

I've tried to apply the patch but am getting the following response. I have a brand new installation of Nextcloud 20.0.6.1 with Ubuntu 20.04.1 LTS and PHP 7.4.14. 

I am a total nub to Nextcloud administration and git, this was my first build so any help will be much appreciated. 

Screenshot 2021-01-28 162446

**Comment by ManafxMT** [comment]
Do besides being a developer you are a magician too 👍💐
First, thanks for your quick response.
Let me understand: if not installed, I need to install "patch".
After that, in the terminal I only put
Patch (downloadpath.patch)?
With or without sudo?
Is there a chance to messup my whole installation- so shall I do a backup before?
I found some videos around the patch command, but still not sure...

**Comment by ChristophWurst** [comment]
`patch -p1 < 25335.patch` should be the command. No sudo required although you possible want to run this as your web server user, like `sudo -u www-data patch …` on Debian/Ubuntu.

**Comment by ManafxMT** [comment]
Turbo response, thanks.
Before I try it: when will be the app available in the app section on 20.06 again? 
Maybe I better wait for it...

**Comment by ChristophWurst** [comment]
it looks like we can release 20.0.7 next week, but I can't promise it

**Comment by ManafxMT** [comment]
Cool, I think I better wait - if there is a slight chance to mess up the patch - I will manage. (joking).
Thanks for your time and explainations...will be very helpful for others for sure.

**Comment by vladlucky** [comment]
Thank your Christoph, the patch resolved the issue and I am now able to install the TOTP plug-in!

**Comment by vladlucky** [comment]
Altough TOTP is working after applying the patch, I started seeing the following errors. In addition to seeing the errors, the app store is not always accessible - sometimes it loads sometimes it doesn't:

image
[appstoreFetcher] Warning: Could not connect to appstore: cURL error 28: Connection timed out after 60000 milliseconds (see https://curl.haxx.se/libcurl/c/libcurl-errors.html)

GET /settings/apps
from 69.248.189.2 by 7256D217-A948-4F55-A6EE-A1D49D2E823B at 2021-01-30T22:53:36-05:00

**Comment by vladlucky** [comment]
Although TOTP is working after applying the patch, I started seeing the following errors. In addition to seeing the errors, the app store is not always accessible - sometimes it loads sometimes it doesn't:

image

image

image

**Comment by ChristophWurst** [comment]
the app store was down, that's why you saw a "new" exception

**Comment by natrius** [comment]
Just to clarify: This seems to resolve the problem with extracting Nextcloud News as well? https://github.com/nextcloud/news/issues/1086

**Comment by ManafxMT** [comment]
I installed NC 20.0 7 and the Email App - all works great, thanks a lot.
It is reacting faster then before too.

Only it is not keeping my signature - can I do something about it?

**Comment by ManafxMT** [comment]
Managed - even with the signature.
Total happy now.

You are doing an awesome job 👍👍

**Comment by ChristophWurst** [comment]
> Just to clarify: This seems to resolve the problem with extracting Nextcloud News as well? [nextcloud/news#1086](https://github.com/nextcloud/news/issues/1086)

https://github.com/nextcloud/news/issues/1086#issuecomment-771914810.



> Only it is not keeping my signature - can I do something about it?

Known regression but feel free to report is as bug for better tracking.

:v:

**Comment by fischer-felix** [comment]
> `patch -p1 < 25335.patch` should be the command. No sudo required although you possible want to run this as your web server user, like `sudo -u www-data patch …` on Debian/Ubuntu.

How would you apply this patch when using a nextcloud snap installation?

**Comment by ChristophWurst** [comment]
> How would you apply this patch when using a nextcloud snap installation?

Please ask on https://help.nextcloud.com/c/support/7

**Comment by DennisWilken** [comment]
Hi Christoph, thanks for your patch. I am trying to install it on a Docker container. But I am not quite sure how to apply it. I downloaded it using `curl https://patch-diff.githubusercontent.com/raw/nextcloud/server/pull/25335.patch -o 25335.patch`
and tried to install it using `patch -p1 < 25335.patch`.  Next, I am asked for the file to patch:

image

Do I have to run the patch in a certain folder? What path do I have to enter? Am I doing anything else wrong? Thanks for your help!

**Comment by szaimen** [comment]
@DennisWilken just update your NC to 20.0.7 (it is included with this version). There is no need to apply the fix manually.

**Comment by DennisWilken** [comment]
> @DennisWilken just update your NC to 20.0.7 (it is included with this version). There is no need to apply the fix manually.

Version 20.0.7 is not yet available on docker (see [here](https://hub.docker.com/_/nextcloud )) So I can either wait, patch or manually upgrade.

**Pull request merged**